runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java

```diff
@@ -209,4 +211,20 @@ void checkForTimeout()
 		final Instant now = Instant.now();
 		events.removeIf(event -> event.getType().isShouldTimeout() && now.isAfter(event.getUpdated().plus(actionTimeout)));
 	}
-}
+
+	public void checkForMenuTimeout()
+	{
+		if (events.isEmpty())
+		{
+			return;
+		}
+		EventWithTime event = events.get(0);
+		if (event.getType().getState().equals(DiscordGameEventType.IN_MENU.getState()))
+		{
+			if (event.start.plusSeconds(MENU_TIMEOUT).isBefore(Instant.now()))
```

Comment:
this should imo respect config.actionTimeout. Maybe it can also just be done in checkForTimeout method?

Reply:
I was thinking about doing that also but wasn't sure. Committed with the changes you requested.